src/Common/tests/System/Security/Cryptography/AlgorithmImplementations/RSA/RSAXml.cs

```diff
@@ -2,19 +2,1461 @@
+                    () => rsa.FromXmlString(
+                        @"
+<RSAKeyValue>
+  <D>
+    r+byNi+cr17FpJH4MCEiPXaKnmkH4c4U52EJtL9yg2gijBrpYkat3c2nWb0EGGi5
+    aWgXxQHoi7z97/ACD4X3KQ==
+  </D>
+  <Unrelated>This is not base64.</Unrelated>
+  <DQ>
+    rCTM8dSbopUADWnD4jArhU50UhWAIaM6ZrKqC8k0RKs=
+  </DQ>
+  <DP>
+    AAiO/cUU9RIt8A2B84gf2ZfuV2nPMaSuZpTPFC/K5Us=
+  </DP>
+  <Q>
+    wLI66OpSqftDTv1KUfNe6+hyoh23ggzUSYiWuVT0Ya8=
+  </Q>
+  <P>
+    83J7HoZ2IFWU08c6AidNTxqYBEopUchczRLG/FetGXs=
+  </P>
+  <Exponent>AQAB</Exponent>
+  <Modulus>
+    tz9Z9e6L1V4kt/8CmtFqhUPJbSU+VDGbk1MsQcPBR3uJ2y0vM9e5qHRYSOBqjmg7
+    UERRHhvKNiUn4Xz0KzgGFQ==
+  </Modulus>
+</RSAKeyValue>
+"));
+            }
+        }
+
+        [Fact]
+        public static void FromXml_BadBase64()
+        {
+            using (RSA rsa = RSAFactory.Create())
+            {
+                // The D value is missing the terminating ==.
+                Assert.Throws<FormatException>(
+                    () => rsa.FromXmlString(
+                        @"
+<RSAKeyValue>
+  <D>
+    r+byNi+cr17FpJH4MCEiPXaKnmkH4c4U52EJtL9yg2gijBrpYkat3c2nWb0EGGi5
+    aWgXxQHoi7z97/ACD4X3KQ
+  </D>
+  <Unrelated>This is not base64.</Unrelated>
+  <InverseQ>
+    wsmVlMaMQHY36wRrMflOgRzNDMqqnu32O4Y1s4+GgQs=
+  </InverseQ>
+  <DQ>
+    rCTM8dSbopUADWnD4jArhU50UhWAIaM6ZrKqC8k0RKs=
+  </DQ>
+  <DP>
+    AAiO/cUU9RIt8A2B84gf2ZfuV2nPMaSuZpTPFC/K5Us=
+  </DP>
+  <Q>
+    wLI66OpSqftDTv1KUfNe6+hyoh23ggzUSYiWuVT0Ya8=
+  </Q>
+  <P>
+    83J7HoZ2IFWU08c6AidNTxqYBEopUchczRLG/FetGXs=
+  </P>
+  <Exponent>AQAB</Exponent>
+  <Modulus>
+    tz9Z9e6L1V4kt/8CmtFqhUPJbSU+VDGbk1MsQcPBR3uJ2y0vM9e5qHRYSOBqjmg7
+    UERRHhvKNiUn4Xz0KzgGFQ==
+  </Modulus>
+</RSAKeyValue>"));
+            }
+        }
+
+        private static void TestReadXml(string xmlString, in RSAParameters expectedParameters)
+        {
+            using (RSA rsa = RSAFactory.Create())
+            {
+                rsa.FromXmlString(xmlString);
+                Assert.Equal(expectedParameters.Modulus.Length * 8, rsa.KeySize);
+
+                bool includePrivateParameters = expectedParameters.D != null;
+
+                ImportExport.AssertKeyEquals(
+                    expectedParameters,
+                    rsa.ExportParameters(includePrivateParameters));
+            }
+        }
+
+        [Fact]
+        public static void FromNullXml()
+        {
+            using (RSA rsa = RSAFactory.Create())
+            {
+                AssertExtensions.Throws<ArgumentNullException>(
+                    "xmlString",
+                    () => rsa.FromXmlString(null));
+            }
+        }
+
+        [Fact]
+        public static void FromInvalidXml()
+        {
+            using (RSA rsa = RSAFactory.Create())
+            {
+                Exception exception = Assert.ThrowsAny<Exception>(
+                    () => rsa.FromXmlString(
+                        @"
+<RSAKeyV
```

Comment:
I'm wondering if this is something we should fix

Reply:
What kind of fix?  Adding a namespace might break someone who is expecting it to not have one.  The wrong namespace, and Base64Binary vs CryptoBinary (which, if fixed, would break import back in netfx) are just characteristics of these methods as deviations from the spec.

I think the design assumed it was being rendered inside an element that already declared the implicit inherited namespace.

Reply:
fine with me, although I think if you try to nest it inside such document (using XML APIs) it will explicitly remove that namespace though